components/script/script_thread.rs

```diff
@@ -1890,20 +1901,28 @@ impl ScriptThread {
 
             let exiting = self.profile_event(category, pipeline_id, move || {
                 match msg {
-                    FromConstellation(ConstellationControlMsg::ExitScriptThread) => {
+                    MixedMessage::FromConstellation(ConstellationControlMsg::ExitScriptThread) => {
                         self.handle_exit_script_thread_msg(can_gc);
                         return true;
                     },
-                    FromConstellation(inner_msg) => {
+                    MixedMessage::FromConstellation(inner_msg) => {
                         self.handle_msg_from_constellation(inner_msg, can_gc)
                     },
-                    FromScript(inner_msg) => self.handle_msg_from_script(inner_msg),
-                    FromDevtools(inner_msg) => self.handle_msg_from_devtools(inner_msg, can_gc),
-                    FromImageCache(inner_msg) => self.handle_msg_from_image_cache(inner_msg),
+                    MixedMessage::FromScript(inner_msg) => self.handle_msg_from_script(inner_msg),
+                    MixedMessage::FromDevtools(inner_msg) => {
+                        self.handle_msg_from_devtools(inner_msg, can_gc)
+                    },
+                    MixedMessage::FromImageCache(inner_msg) => {
+                        self.handle_msg_from_image_cache(inner_msg)
+                    },
                     #[cfg(feature = "webgpu")]
-                    FromWebGPUServer(inner_msg) => {
+                    MixedMessage::FromWebGPUServer(inner_msg) => {
                         self.handle_msg_from_webgpu_server(inner_msg, can_gc)
                     },
+                    MixedMessage::TimerFired => self
```

Comment:
I think what is needed to fix those timeouts is to call `dispatch_completed_timers` not only when `MixedMessage::TimerFired` is received, but always [here](https://github.com/servo/servo/blob/a21616531c69fd7d59c649bbfb67d247fb54f00e/components/script/script_thread.rs#L1734).

Because:
1. Something else may repeatedly wake-up the `select!`, so you may actually never, or only very late, run this branch.
2. Even if the `MixedMessage::TimerFired` is received immediately, then the actual timer tasks are only scheduled here, which means it will always take one more iteration of `handle_msgs` for them to be run. 

So by always running `dispatch_completed_timers` before the inner loop of `handle_msgs` that drains the message queue and puts all script tasks in `sequential`, you ensure that pending timers ready to fire at that point are added to `sequential` and run alongside other script-tasks. In fact we're getting a new nice property which is that they are all coalesced together. 

The channel returned by `wait_channel` is only there for the case where there is nothing else waking up the script-thread.